code/game/objects/items/devices/flashlight.dm

```diff
@@ -20,8 +20,10 @@
 /obj/item/flashlight/update_icon_state()
 	if(on)
 		icon_state = "[initial(icon_state)]-on"
+		item_state = "[initial(item_state)]-on"
 	else
 		icon_state = "[initial(icon_state)]"
+		item_state = "[initial(item_state)]"
```

Comment:
This causes flashlights that dont have a -on item_state to go invisible, you should probably scope it to only those that have it.